### ISSUE TITLE
Running nightly clippy is generating a lifetime warning for a derive macro.

### ISSUE THREAD

**Issue by BlackDex** [diesel-rs/diesel#3182]
## Setup
Using the examples within this repository (And also on the Vaultwarden codebase) running the nightly clippy generates a warning regarding lifetimes. While this isn't a huge issue in general, and the code will build just fine, some CI/CD tooling might break because of this if clippy generates a warning or error regarding this.

Also adding `#[allow(clippy::extra_unused_lifetimes)]` above or below that derive will not work, because that is not where the lifetimes are set since they happen within the macro.

### Versions

- **Rust:** rustc 1.63.0-nightly (9257f5aad 2022-05-21)
- **Cargo:** cargo 1.63.0-nightly (a4c1cd0eb 2022-05-18)
- **Diesel:** 4fe987676e7c465f2fdc116fdf11ba4378e64f9a
- **Database:** postgres, mysql, sqlite
- **Operating System:** Linux

### Feature Flags

- **diesel:** the respective database, nothing else.

## Problem Description

When for example using the `getting_started_step_3` example on all the database types running the nightly clippy will generate a warning.

```
    Checking diesel_demo_step_3_sqlite v0.1.0 (/home/mathijs/rust/diesel/examples/sqlite/getting_started_step_3)
warning: this lifetime isn't used in the impl
   --> examples/sqlite/getting_started_step_3/src/models.rs:12:10
    |
12  | #[derive(Insertable)]
    |          ^^^^^^^^^^ in this derive macro expansion
    |
   ::: /home/mathijs/rust/diesel/diesel_derives/src/lib.rs:403:1
    |
403 | pub fn derive_insertable(input: TokenStream) -> TokenStream {
    | ----------------------------------------------------------- in this expansion of `#[derive(Insertable)]`
    |
    = note: `#[warn(clippy::extra_unused_lifetimes)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#extra_unused_lifetimes
```

If i run this on the Vaultwarden codebase it also reports the same for the `AsChangeset` derive.

### What are you trying to accomplish?

Running the nightly clippy. I do this once in a while to check newly generated clippy results before they might occur when a new release of rust is generated.

### What is the expected output?

No clippy errors from the macro generated code.

### What is the actual output?

Clippy errors from the macro generated code.

### Are you seeing any additional errors?

There are some other clippy errors on the example using both current stable (`cargo 1.61.0 (a028ae4 2022-04-29)`) and nightly clippy.

### Steps to reproduce

- Go to one of the database type examples like `examples/sqlite/getting_started_step_3`
- Run clippy with either stable or nightly `cargo +nightly clippy -- -Z macro-backtrace` or `cargo +stable clippy`
- Check the warnings.

## Checklist

- [x] I have already looked over the [issue tracker](https://github.com/diesel-rs/diesel/issues)  and the [disussion forum](https://github.com/diesel-rs/diesel/discussions)  for similar possible closed issues.
 
- [x] This issue can be reproduced on Rust's stable channel. (Your issue will be
  closed if this is not the case)
- [x] This issue can be reproduced without requiring a third party crate

**Comment by weiznich** [comment]
Fixed with #3185

**Comment by BlackDex** [comment]
Nice thanks @weiznich ! 🎉
